### PR TITLE
LPS 182364 general fix to delete entities owned by the user and not all the entities the user has touch

### PR DESCRIPTION
**Pull request by robertoDiaz**

### CODE REVIEW

**Comment by liferay-lima** [comment]
ci:test:relevant

**Comment by robertoDiaz** [comment]
ci:test:relevant

File: modules/apps/portal-language/portal-language-lang/src/main/resources/content/Language.properties [hunk 1]
```diff
@@ -18240,6 +18240,7 @@ thousands-delimiter=,
 thousands-separator=Thousands Separator
 thread=Thread
 thread-as-question-by-default=Thread as Question by Default
+thread-cannot-be-deleted.-anonymimze-instead=Thread cannot be deleted. Anonyimize instead.
```
Comment (adolfopa) [inline modules/apps/portal-language/portal-language-lang/src/main/resources/content/Language.properties:18243]: There's a typo here, both in the key and the value (anonymimze / Anonyimize)

**Review by adolfopa — CHANGES_REQUESTED** [review]

File: modules/apps/portal-language/portal-language-lang/src/main/resources/content/Language.properties (hunk 1 above)
Comment (robertoDiaz) [reply modules/apps/portal-language/portal-language-lang/src/main/resources/content/Language.properties:18243]: Fixed

File: modules/apps/portal-language/portal-language-lang/src/main/resources/content/Language.properties [hunk 2]
```diff
@@ -18240,6 +18240,7 @@ thousands-delimiter=,
 thousands-separator=Thousands Separator
 thread=Thread
 thread-as-question-by-default=Thread as Question by Default
+thread-cannot-be-deleted.-anonymize-instead=Thread cannot be deleted. Anonymize instead.
```
Comment (adolfopa) [inline modules/apps/portal-language/portal-language-lang/src/main/resources/content/Language.properties:18243]: @austinchiang What do you think about this piece of text? In particular, the "Anonymize instead" part sounds a bit weird to me, but I'm not sure.

Comment (austinchiang) [reply modules/apps/portal-language/portal-language-lang/src/main/resources/content/Language.properties:18243]: Maybe "The thread cannot be deleted. Anonymize it instead." or "The thread cannot be deleted. Please anonymize it."

File: modules/apps/portal-language/portal-language-lang/src/main/resources/content/Language.properties [hunk 3]
```diff
@@ -18240,7 +18240,7 @@ thousands-delimiter=,
 thousands-separator=Thousands Separator
 thread=Thread
 thread-as-question-by-default=Thread as Question by Default
-thread-cannot-be-deleted.-anonymize-instead=Thread cannot be deleted. Anonymize instead.
+thread-cannot-be-deleted.-anonymize-it-instead=Thread cannot be deleted. Anonymize it instead.
```
Comment (adolfopa) [inline modules/apps/portal-language/portal-language-lang/src/main/resources/content/Language.properties:18243]: Can you move this change outside of the `build lang` commit? This is not autogenerated.

**Review by adolfopa — CHANGES_REQUESTED** [review]

**Review by adolfopa — APPROVED** [review]

**Comment by adolfopa** [comment]
@robertoDiaz Changes look fine, but there are rebase conflicts.

**Comment by robertoDiaz** [comment]
I'll resolve them, update and do the forward, if you don't mind :)

**Comment by robertoDiaz** [comment]
I've resolve the issues.

running ci:forward

**Comment by robertoDiaz** [comment]
ci:forward

**Comment by Hanlf** [comment]
ci:test:relevant

**Comment by Hanlf** [comment]
Hi @robertoDiaz 
There is a compile failure for this PR, so no test run on the ci:test:relevant job, could you take a look?
```
     [exec] /opt/dev/projects/github/liferay-portal/modules/apps/object/object-service/src/main/java/com/liferay/object/internal/uad/anonymizer/ObjectEntryUADAnonymizer.java:36: error: com.liferay.object.internal.uad.anonymizer.ObjectEntryUADAnonymizer is not abstract and does not override abstract method delete(com.liferay.object.model.ObjectEntry,long) in com.liferay.user.associated.data.anonymizer.UADAnonymizer
     [exec] public class ObjectEntryUADAnonymizer
     [exec]        ^
     [exec] /opt/dev/projects/github/liferay-portal/modules/apps/object/object-service/src/main/java/com/liferay/object/internal/uad/anonymizer/ObjectEntryUADAnonymizer.java:95: error: method does not override or implement a method from a supertype
     [exec] 	@Override
     [exec] 	^
     [exec] 
     [exec] > Task :apps:object:object-web:compileJava
     [exec] 
     [exec] > Task :apps:object:object-web:copyLibs SKIPPED
     [exec] > Task :apps:object:object-web:copyCSS SKIPPED
     [exec] > Task :apps:object:object-web:buildCSS SKIPPED
     [exec] 
     [exec] > Task :apps:object:object-service:compileJava
     [exec] Note: Some input files use or override a deprecated API.
     [exec] Note: Recompile with -Xlint:deprecation for details.
     [exec] Note: Some input files use unchecked or unsafe operations.
     [exec] Note: Recompile with -Xlint:unchecked for details.
     [exec] 2 errors
     [exec] 
     [exec] > Task :apps:object:object-service:compileJava FAILED
```

**Comment by robertoDiaz** [comment]
Sure

**Comment by robertoDiaz** [comment]
Updated (seems that this class was commited the last weeks)

**Comment by robertoDiaz** [comment]
ci:forward

**Comment by Hanlf** [comment]
ci:test:relevant

**Comment by Hanlf** [comment]
ci:test:relevant

**Comment by Hanlf** [comment]
ci:test:relevant

**Comment by Hanlf** [comment]
ci:test:relevant

**Comment by Hanlf** [comment]
Hi @robertoDiaz 
There are some failures related to this PR, could you take a look?

- Failire1: These tests all failed at "No MBDiscussion exists with the key {threadId=*****}"

[com.liferay.message.boards.search.test.MBMessageSearchTest](https://testray.liferay.com/en/home/-/testray/case_results/2342630050?redirect=https%3A%2F%2Ftestray.liferay.com%2Fzh%2Fhome%2F-%2Ftestray%2Fcase_results%2Findex%3FtestrayBuildId%3D2342609311%26testrayRunId%3D2342610790%26cur%3D1%26delta%3D200%26orderByCol%3Dstatus_sortable%26orderByType%3Dasc)
[com.liferay.document.library.uad.anonymizer.test.DLFileEntryUADAnonymize...](https://testray.liferay.com/en/home/-/testray/case_results/2342629764?redirect=https%3A%2F%2Ftestray.liferay.com%2Fzh%2Fhome%2F-%2Ftestray%2Fcase_results%2Findex%3FtestrayBuildId%3D2342609311%26testrayRunId%3D2342610790%26cur%3D1%26delta%3D200%26orderByCol%3Dstatus_sortable%26orderByType%3Dasc)
[com.liferay.document.library.uad.display.test.DLFolderUADDisplayTest](https://testray.liferay.com/en/home/-/testray/case_results/2342629782?redirect=https%3A%2F%2Ftestray.liferay.com%2Fzh%2Fhome%2F-%2Ftestray%2Fcase_results%2Findex%3FtestrayBuildId%3D2342609311%26testrayRunId%3D2342610790%26cur%3D1%26delta%3D200%26orderByCol%3Dstatus_sortable%26orderByType%3Dasc)
[com.liferay.message.boards.exportimport.test.MBExportImportTest](https://testray.liferay.com/en/home/-/testray/case_results/2342629971?redirect=https%3A%2F%2Ftestray.liferay.com%2Fzh%2Fhome%2F-%2Ftestray%2Fcase_results%2Findex%3FtestrayBuildId%3D2342609311%26testrayRunId%3D2342610790%26cur%3D1%26delta%3D200%26orderByCol%3Dstatus_sortable%26orderByType%3Dasc)
[com.liferay.message.boards.service.test.MBThreadLocalServiceTest](https://testray.liferay.com/en/home/-/testray/case_results/2342630307?redirect=https%3A%2F%2Ftestray.liferay.com%2Fzh%2Fhome%2F-%2Ftestray%2Fcase_results%2Findex%3FtestrayBuildId%3D2342609311%26testrayRunId%3D2342610790%26cur%3D1%26delta%3D200%26orderByCol%3Dstatus_sortable%26orderByType%3Dasc)
[com.liferay.message.boards.uad.display.test.MBMessageUADDisplayTest](https://testray.liferay.com/en/home/-/testray/case_results/2342630259?redirect=https%3A%2F%2Ftestray.liferay.com%2Fzh%2Fhome%2F-%2Ftestray%2Fcase_results%2Findex%3FtestrayBuildId%3D2342609311%26testrayRunId%3D2342610790%26cur%3D1%26delta%3D200%26orderByCol%3Dstatus_sortable%26orderByType%3Dasc)

- Failure2: [com.liferay.message.boards.internal.moderation.test.MBModerationTest](https://testray.liferay.com/zh/home/-/testray/case_results/2342629983?redirect=https%3A%2F%2Ftestray.liferay.com%2Fzh%2Fhome%2F-%2Ftestray%2Fcase_results%2Findex%3FtestrayBuildId%3D2342609311%26testrayRunId%3D2342610790%26cur%3D1%26delta%3D200%26orderByCol%3Dstatus_sortable%26orderByType%3Dasc)

- Failure3: [PortalLogAssertorTest-modules-integration-mysql57-jdk8/1/20](https://testray.liferay.com/zh/home/-/testray/case_results/2342630127?redirect=https%3A%2F%2Ftestray.liferay.com%2Fzh%2Fhome%2F-%2Ftestray%2Fcase_results%2Findex%3FtestrayBuildId%3D2342609311%26testrayRunId%3D2342610790%26cur%3D1%26delta%3D200%26orderByCol%3Dstatus_sortable%26orderByType%3Dasc)

- Failure4: [com.liferay.headless.delivery.resource.v1_0.test.MessageBoardMessageReso...](https://testray.liferay.com/zh/home/-/testray/case_results/2342629946?redirect=https%3A%2F%2Ftestray.liferay.com%2Fzh%2Fhome%2F-%2Ftestray%2Fcase_results%2Findex%3FtestrayBuildId%3D2342609311%26testrayRunId%3D2342610790%26cur%3D1%26delta%3D200%26orderByCol%3Dstatus_sortable%26orderByType%3Dasc)

- Failure5: [com.liferay.headless.delivery.resource.v1_0.test.MessageBoardAttachmentR...](https://testray.liferay.com/zh/home/-/testray/case_results/2342629943?redirect=https%3A%2F%2Ftestray.liferay.com%2Fzh%2Fhome%2F-%2Ftestray%2Fcase_results%2Findex%3FtestrayBuildId%3D2342609311%26testrayRunId%3D2342610790%26cur%3D1%26delta%3D200%26orderByCol%3Dstatus_sortable%26orderByType%3Dasc)

- Failure6: [com.liferay.message.boards.trash.test.MBThreadTrashHandlerTest](https://testray.liferay.com/zh/home/-/testray/case_results/2342630188?redirect=https%3A%2F%2Ftestray.liferay.com%2Fzh%2Fhome%2F-%2Ftestray%2Fcase_results%2Findex%3FtestrayBuildId%3D2342609311%26testrayRunId%3D2342610790%26cur%3D1%26delta%3D200%26orderByCol%3Dstatus_sortable%26orderByType%3Dasc)

- Failure7: [com.liferay.message.boards.uad.exporter.test.MBMessageUADExporterTest](https://testray.liferay.com/zh/home/-/testray/case_results/2342630291?redirect=https%3A%2F%2Ftestray.liferay.com%2Fzh%2Fhome%2F-%2Ftestray%2Fcase_results%2Findex%3FtestrayBuildId%3D2342609311%26testrayRunId%3D2342610790%26cur%3D1%26delta%3D200%26orderByCol%3Dstatus_sortable%26orderByType%3Dasc)

- Failure8: [com.liferay.message.boards.uad.exporter.test.MBThreadUADExporterTest](https://testray.liferay.com/zh/home/-/testray/case_results/2342630306?redirect=https%3A%2F%2Ftestray.liferay.com%2Fzh%2Fhome%2F-%2Ftestray%2Fcase_results%2Findex%3FtestrayBuildId%3D2342609311%26testrayRunId%3D2342610790%26cur%3D1%26delta%3D200%26orderByCol%3Dstatus_sortable%26orderByType%3Dasc)

**Comment by robertoDiaz** [comment]
Sure. I'll check it this afternoon and resend with the fixes.

**Comment by robertoDiaz** [comment]
Stilll checking the issues.

I'm going to retest to move forward (some of the found issues do not make sense to me)

**Comment by robertoDiaz** [comment]
ci:test:relevant

**Comment by robertoDiaz** [comment]
Most of test are related with the MB discussion removal logic.

I've updated it an seems to work in most cases.

rerunning the tests

**Comment by robertoDiaz** [comment]
ci:test:relevant

**Comment by adolfopa** [comment]
@robertoDiaz Putting this "on hold" until it is ready so that nobody else tries to do anything with it.

**Comment by robertoDiaz** [comment]
ci:reevaluate:1287401_6693

**Comment by robertoDiaz** [comment]
> @robertoDiaz Putting this "on hold" until it is ready so that nobody else tries to do anything with it.

I'm retesting with a rebased code and if it fails again, I'll proceed as agreed in slack.

**Comment by robertoDiaz** [comment]
ci:test:relevant

**Comment by robertoDiaz** [comment]
ci:test:relevant

**Comment by robertoDiaz** [comment]
I'll resend with updated changes.

**Pull request closed without merging**